resilience4j/tests/test_resilience4j.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+
+from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.resilience4j import Resilience4jCheck
+
+EXPECTED_PROMETHEUS_METRICS = {
+    "bulkhead.available.concurrent.calls",
+    "bulkhead.max.allowed.concurrent.calls",
+    "bulkhead.max.thread.pool.size",
+    "bulkhead.queue.capacity",
+    "bulkhead.queue.depth",
+    "bulkhead.thread.pool.size",
+    "circuitbreaker.buffered.calls",
+    "circuitbreaker.calls.seconds.count",
+    "circuitbreaker.calls.seconds.max",
+    "circuitbreaker.calls.seconds.sum",
+    "circuitbreaker.calls",
+    "circuitbreaker.failure.rate",
+    "circuitbreaker.max.buffered.calls",
+    "circuitbreaker.state",
+    "ratelimiter.available.permissions",
+    "ratelimiter.waiting.threads",
+    "retry.calls",
+    "timelimiter.calls",
+}
+
+
+@pytest.mark.integration
+def test_check(dd_run_check, aggregator, check, mock_prometheus_metrics):
+    dd_run_check(check)
+    for metric_name in EXPECTED_PROMETHEUS_METRICS:
+        aggregator.assert_metric(metric_name, at_least=0)
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+    aggregator.assert_service_check("resilience4j.prometheus.health", status=Resilience4jCheck.OK)
```

Comment:
Let's add couple unit tests (in a separate file) for the happy path and for the error path (wrong openmetrics_endpoint setup for instance).

[Example](https://github.com/DataDog/integrations-core/blob/master/teleport/tests/test_common.py)

Reply:
Thank you for the example. I did some changes, could you check please?

Reply:
Nice! The `@pytest.mark.integration` is no longer necessary as the pymark applies to the whole file.
With that, it would be nice to add a `test_unit.py` file with some unit tests same as in the example above, and rename the existing `test_resilience4j.py` to `test_integration.py.

The `check.py` previous implementation was good, we could revert to that one.

Reply:
I followed the suggestions:
1. Reverting the `check.py`
2. Renaming `test_resilience4j.py` to `test_integration.py`
3. Adding `test_unit.py` (I hope is it okay)